src/openms/source/FORMAT/XMLFile.cpp

```diff
@@ -152,21 +154,38 @@ namespace OpenMS
       try
       {
         parser->parse(*source);
-        delete(parser);
+        delete parser;
         delete source;
       }
       catch (const xercesc::XMLException & toCatch)
       {
-        throw Exception::ParseError(__FILE__, __LINE__, __PRETTY_FUNCTION__, "", String("XMLException: ") + StringManager().convert(toCatch.getMessage()));
+        delete parser;
+        delete source;
+        throw Exception::ParseError(__FILE__, __LINE__, __PRETTY_FUNCTION__, "", 
+            String("XMLException: ") + StringManager().convert(toCatch.getMessage()));
       }
       catch (const xercesc::SAXException & toCatch)
       {
-        throw Exception::ParseError(__FILE__, __LINE__, __PRETTY_FUNCTION__, "", String("SAXException: ") + StringManager().convert(toCatch.getMessage()));
+        delete parser;
+        delete source;
+        throw Exception::ParseError(__FILE__, __LINE__, __PRETTY_FUNCTION__, "",
+            String("SAXException: ") + StringManager().convert(toCatch.getMessage()));
       }
       catch (const XMLHandler::EndParsingSoftly & /*toCatch*/)
       {
-        //nothing to do here, as this exception is used to softly abort the parsing for whatever reason.
+        // nothing to do here except free memory, as this exception is used to
+        // softly abort the parsing for whatever reason.
+        delete parser;
+        delete source;
```

Comment:
agreed, good catch!

A clean alternative to all these delete calls, would be RAII, i.e. either a custom class (as done in  XMLCleaner_) or an auto_ptr/std::unique_ptr. The former is not encouraged for C11 and I'm not sure about the broad support of std::unique_ptr for the platforms we run on. VS2010 does have it though.